### PR TITLE
Add user guide and developer guide

### PR DESCRIPTION
**Pull request by AsiaCao**
Polish documentation:

1. Add user guide
2. Add developer guide
3. Tweak the format of a few API docs
4. Fix an expired link in one example code

### CODE REVIEW

**Review by andy-ldai — APPROVED** [review]
looks good

File: README.md [hunk 1]
```diff
@@ -65,82 +64,21 @@ image_with_preds = overlay_predictions(predictions, image)
...
-poetry shell
-```
-
 ## Documentation
 
 -  [Landing AI Python Library API Reference](https://landing-ai.github.io/landingai-python/landingai.html)
-
+-  [Landing AI Python Library User Guide](https://landing-ai.github.io/landingai-python/landingai.html#user-guide)
+-  [Landing AI Python Library Developer Guide](https://landing-ai.github.io/landingai-python/landingai.html#developer-guide)
 -  [Landing AI Suport Center](https://support.landing.ai/)
```
Comment (lai-bianca) [inline README.md:83]: Line 83: Change "Suport" to "Support" (that was my typo, sorry!)

**Review by lai-bianca — COMMENTED** [review]
I left one comment on the main README file. Other than that, looks good!

File: README.md (hunk 1 above)
Comment (AsiaCao) [reply README.md:83]: good catch, done: https://github.com/landing-ai/landingai-python/commit/ca9866fdeb2c9d603c1994813977a45f4cfe12c5

**Pull request merged**